andela/ah-leagueOfLegends
# #159965480 Users should be able to create their profiles

## What does this PR do?
```
- Enables users to create and update their profiles and their profiles.
- It also enables other users to view other peoples profile.
```

## Description of Task to be completed?
```
- Implement Profile object creation upon creating a user
- Create a  URL  and its view that a allows a user's profile to be viewed
- Create an exception handler for users not found
- Create a method to generate and Avatar for a user profile
- Write unit  tests for the added features

```
## How should this be manually tested?


- clone the repo 
- checkout ft-profiles-159965480 branch 
- Register a user by sending a POST request to  ```http://127.0.0.1:8000/api/users/```
- Then check whether the user profile associated with it is created by sending a GET request to 
   ```http://127.0.0.1:8000/api/profiles/<username>```
- To edit update user data, which shall be reflected in the profile. Go to  ```http://127.0.0.1:8000/api/user/``` and change the desired fields. 
An example:
```
{
  "user": {
     "bio": "I bake ",
    "image":" https://vsco.co/tichtoo/media/544f672f7367089c0c8b4606"
  }
}
```
## What are the relevant pivotal tracker stories?

[#159965480](https://www.pivotaltracker.com/story/show/159965480)
